packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -838,3 +838,13 @@ function endsWith(string, suffix) {
     return string.endsWith(suffix);
   }
 }
+
+Ember.defineProperty(RESTAdapter, 'maxUrlLength', {
```

Comment:
Here is the code Ember uses to deprecate a property. https://github.com/emberjs/ember.js/blob/1b5adaa65d20af4f4e57702ea34aa0dcc42e1489/packages/ember-metal/lib/deprecate_property.js#L24-L43. It seems to work even if a user overrides this getter. Its also usually called on the classes's `prototype` object.